clients/src/main/java/org/apache/kafka/clients/Metadata.java

```diff
@@ -331,6 +338,23 @@ public synchronized void removeListener(Listener listener) {
         this.listeners.remove(listener);
     }
 
+    /**
+     * "Close" this metadata instance to indicate that metadata updates are no longer possible. This is typically used
+     * when the thread responsible for performing metadata updates is exiting and needs a way to relay this information
+     * to any other thread(s) that could potentially wait on metadata update to come through.
+     */
+    public synchronized void close() {
```

Comment:
Probably not a big deal since `awaitUpdate` will not block in `wait` for longer than 100ms, but maybe we may as well call `notifyAll()`?

Reply:
Should Metadata extend Closeable?